app/src/main/java/com/vultisig/wallet/data/repositories/ChainsOrderRepository.kt

```diff
@@ -0,0 +1,107 @@
+package com.vultisig.wallet.data.repositories
+
+import com.vultisig.wallet.data.db.ChainOrderDao
+import com.vultisig.wallet.data.db.models.ChainOrderEntity
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+
+internal interface ChainsOrderRepository {
+    fun loadByOrders(): Flow<List<ChainOrderEntity>>
+    suspend fun updateItemOrder(lowerVal: String?, middleVal: String, upperVal: String?)
+    suspend fun delete(name: String)
+    suspend fun find(name: String): ChainOrderEntity
+    suspend fun insert(name: String)
+    suspend fun findMaxOrder(): ChainOrderEntity?
+}
+
+
+internal class ChainsOrderRepositoryImpl @Inject constructor(
+    private val chainOrderDao: ChainOrderDao
+) : ChainsOrderRepository {
+    override fun loadByOrders(): Flow<List<ChainOrderEntity>> {
+        return chainOrderDao.loadByOrders()
+    }
+
+    override suspend fun updateItemOrder(lowerVal: String?, middleVal: String, upperVal: String?) {
+        withContext(IO) {
+            val lowerChainEntity = if (lowerVal != null)
+                chainOrderDao.find(lowerVal)
+            else ChainOrderEntity(trueFraction = "0/1", fractionInFloat = 0f)
+            val middleChainEntity = chainOrderDao.find(middleVal)
+            val upperChainEntity = if (upperVal != null) chainOrderDao.find(upperVal)
+            else {
+                val maxOrder = findMaxOrder() ?: ChainOrderEntity(
+                    "", "0/1", 0f
+                )
+                val maxOrderFraction = maxOrder.trueFraction.extractFraction()
+                ChainOrderEntity(
+                    trueFraction = (
+                            maxOrderFraction.first + maxOrderFraction.second
+                                    to maxOrderFraction.second)
+                        .toFractionString(),
+                    fractionInFloat = maxOrder.fractionInFloat + 1
+                )
+            }
+            val (lowerNumerator, lowerDenominator) =
+                lowerChainEntity.trueFraction.extractFraction()
+            val (upperNumerator, upperDenominator) =
+                upperChainEntity.trueFraction.extractFraction()
+            val newMiddleTrueFractionPair =
+                (lowerNumerator + upperNumerator) to (lowerDenominator + upperDenominator)
+            val newMiddleFloatFraction: Float = newMiddleTrueFractionPair.first.toFloat()
+                .div(newMiddleTrueFractionPair.second.toFloat())
+            chainOrderDao.updateItemOrder(
+                middleChainEntity.copy(
+                    trueFraction = newMiddleTrueFractionPair.toFractionString(),
+                    fractionInFloat = newMiddleFloatFraction
+                )
+            )
+        }
```

Comment:
Why do all this stuff with "true fraction"? Why don't just insert an item with idk diff half a diff between floats before and after it? Or maybe something even simpler. This seems excessive.

Reply:
@YGolybev 
I tried to implement the rational in sqllite. (https://begriffs.com/posts/2018-03-20-user-defined-order.html)
If you believe float would be enough for us for now, then I can change it easily

Reply:
Well, I mean, if it's going to work the same but require less code, I would go for it, but it's up to you.